### PR TITLE
DCOS-18171: Add zone, region in the task Table and Details Page

### PR DESCRIPTION
**Pull request by bstavroulakis**
Add zone, region in the task Table and Details Page.

Closes DCOS-18171, DCOS-19136, DCOS-19137, DCOS-19138

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [x] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: plugins/services/src/js/containers/tasks/TaskTable.js [hunk 1]
```diff
@@ -296,6 +317,38 @@ class TaskTable extends React.Component {
     );
   }
 
+  renderRegion(prop, task) {
+    if (!task.region) {
+      return "Primary";
```
Comment (orlandohohmeier) [inline plugins/services/src/js/containers/tasks/TaskTable.js]: @bstavroulakis Let's use the same fallback as in [`NodesTable.js#L119`](https://github.com/dcos/dcos-ui/blob/master/plugins/nodes/src/js/components/NodesTable.js#L119) to be consistent. /cc @Poltergeist

**Comment by orlandohohmeier** [comment]
@bstavroulakis Please squash the commits to ensure they work independently of any later commits.

File: plugins/services/src/js/containers/tasks/TaskTable.js (hunk 1 above)
Comment (bstavroulakis) [reply plugins/services/src/js/containers/tasks/TaskTable.js]: @orlandohohmeier @Poltergeist I'll have a definite answer from product to see what to do here.

File: src/js/structs/CompositeState.js [hunk 2]
```diff
@@ -103,6 +104,10 @@ class CompositeState {
   getNodesList() {
     return new NodesList({ items: this.data.slaves });
   }
+
+  getNodeMaster() {
+    return new Node(this.data.leader_info);
```
Comment (nLight) [inline src/js/structs/CompositeState.js]: We're missing this field in V1 😱 
There's a ticket for the mesos team https://issues.apache.org/jira/browse/MESOS-8078

Comment (nLight) [reply src/js/structs/CompositeState.js]: @dcos/frontend-foundation

Comment (MatApple) [reply src/js/structs/CompositeState.js]: @nLight have you spoken to Vinod about getting this prioritized?

Comment (nLight) [reply src/js/structs/CompositeState.js]: Haven't spoken in person, I pinged him on slack, no reply though.

Comment (nLight) [reply src/js/structs/CompositeState.js]: Ok, so _there is_ a call for that.

`GET_MASTER` call gives the same info and it also should include region and zone info if that master is configured in a such way.

As @vinodkone told me, in V1 only leading master responds to request, so the info we get via `GET_MASTER` is equal to `leader_info`

Comment (nLight) [reply src/js/structs/CompositeState.js]: Please continue using `leader_info`, I will pick it up from three.

**Comment by bstavroulakis** [comment]
Tested! Works as expected, with responsive priority column rules as well.

File: plugins/services/src/js/utils/TaskUtil.js [hunk 3]
```diff
@@ -112,6 +113,71 @@ const TaskUtil = {
...
     });
+  },
+
+  getRegionName(task) {
+    const node = this.getNode(task);
+    if (!node) {
+      return "(Local)";
+    }
+
+    const masterNode = CompositeState.getMasterNode();
```
Comment (orlandohohmeier) [inline plugins/services/src/js/utils/TaskUtil.js]: Not blocking as this is following existing "patterns" but we should try hard to write _pure_ utils that don't access any kind of data "store" directly.

**Comment by orlandohohmeier** [comment]
Im going to squash merge this PR as the commit message don't follow our guidelines.

**Pull request merged**